smac/intensification/intensification.py

```diff
@@ -137,94 +190,362 @@ def __init__(self, tae_runner: ExecuteTARun,
+        # This new challenger is also assumed to be the incumbent
         if self.stage == IntensifierStage.RUN_FIRST_CONFIG:
             if incumbent is None:
-                self.logger.info("First run, no incumbent provided; challenger is assumed to be the incumbent")
+                self.logger.info(
+                    "First run, no incumbent provided;"
+                    " challenger is assumed to be the incumbent"
+                )
+                challenger, self.new_challenger = self.get_next_challenger(
+                    challengers=challengers,
+                    chooser=chooser,
+                )
                 incumbent = challenger
             else:
-                inc_runs = run_history.get_runs_for_config(incumbent, only_max_observed_budget=True)
+                inc_runs = run_history.get_runs_for_config(
+                    incumbent,
+                    only_max_observed_budget=True
+                )
                 if len(inc_runs) > 0:
-                    self.logger.debug("Skipping RUN_FIRST_CONFIG stage since incumbent is already run before")
+                    self.logger.debug(
+                        "Skipping RUN_FIRST_CONFIG stage since "
+                        "incumbent has already been ran"
+                    )
                     self.stage = IntensifierStage.RUN_INCUMBENT
 
+        # LINES 3-7
+        if self.stage in [IntensifierStage.RUN_FIRST_CONFIG, IntensifierStage.RUN_INCUMBENT]:
+
+            # Line 3
+            # A modified version, that not only checks for maxR
+            # but also makes sure that there are runnable instances,
+            # that is, instances has not been exhausted
+            inc_runs = run_history.get_runs_for_config(
+                incumbent,
+                only_max_observed_budget=True
+            )
+
+            # Line 4
+            available_insts = self._get_inc_available_inst(incumbent, run_history)
+            if available_insts and len(inc_runs) < self.maxR:
+                # Lines 5-6-7
+                instance, seed, cutoff = self._get_next_inc_config(available_insts)
+
+                instance_specific = "0"
+                if instance is not None:
+                    instance_specific = self.instance_specifics.get(instance, "0")
+
+                return RunInfoIntent.RUN, RunInfo(
+                    config=incumbent,
+                    instance=instance,
+                    instance_specific=instance_specific,
+                    seed=seed,
+                    cutoff=cutoff,
+                    capped=False,
+                    budget=0.0,
+                )
+            else:
+                # This point marks the transitions from lines 3-7
+                # to 8-18.
+
+                self.logger.debug(
+                    "No further instance-seed pairs for incumbent available."
+                )
+
+                self.stage = IntensifierStage.RUN_CHALLENGER
+
+                self._process_inc_run(
```

Comment:
Shouldn't this have happened in the previous iteration in `process_run`?

Reply:
I completely agree and self._process_inc_run can be removed in favor of simplicity. 

It was only here to print the incumbent performance for debugging purposes.